### ISSUE TITLE
Feature Request: Start music stream from any position

### ISSUE THREAD

**Issue by khers** [sentriz/gonic#483]
gonic version: 0.16.3

if from docker, docker tag: sentriz/gonic:latest

I would like to be able to start a music stream at a specified point in the track. When I pass an offset to the stream endpoint using the `timeOffest` parameter, I get the track from the beginning. My first question is if I am going something wrong with that request? And second, if the request is correct, would you be open to changing the endpoint to honor this parameter for music?

**Comment by sentriz** [comment]
it should work already, with timeOffset not timeOffest

**Comment by sentriz** [comment]
also a transcode rule would need to be active, as in not serving the raw file. but i may change that

**Comment by khers** [comment]
That is why it isn't working for me, I am not transcoding. I'd like to vote for this working on raw files as well but don't have the go chops to make it happen.

**Comment by dweymouth** [comment]
> also a transcode rule would need to be active, as in not serving the raw file. but i may change that

I think you should change that because otherwise it may break clients that use transcodeOffset. (Of which Symfonium is the only one at the moment I think). A client can't know if it's being served a transcoded file, so a client that supports transcodeOffset will probably use it for every seek request, even if you're serving the original file.

**Comment by sentriz** [comment]
what is transcodeOffset?

**Comment by dweymouth** [comment]
OpenSubsonic clients won't use the timeOffset parameter unless you report support of the "transcodeOffset" extension in the getOpenSubsonicExtensions API

**Comment by sentriz** [comment]
hmm it seems navidrome and lms also advertise the extension but drop `timeOffset` on the raw branch

is this an issue on all servers?

**Comment by khers** [comment]
To the best of my knowledge, yes. I thought I had opened a discussion on the API repo about how a client that is not transcoding is supposed to handle seeking and the suggestion was to use this field even when not transcoding. However, I cannot seem to find that discussion now.

 Personally, I don't have a strong preference how it is done (this field, HTTP range requests, or something else) but I do think we should have a way to start streaming at user specified points in any track, regardless of transcoding.

**Comment by sentriz** [comment]
i check the servers, it seems LMS, navidrome, and gonic. all support HTTP Range when format=raw, and timeOffset otherwise. maybe that's enough in this case?

**Comment by khers** [comment]
Yes, I didn't realize that gonic supported range requests. I think this is covered in that case.

We should probably find a way to unify the seek options because a client will not know if transcoding is happening for a stream request.
